sdk/PrebidDemandSDKLoaders/DemandSDKLoaders/FacebookAudienceNetwork/PBFacebookBannerAdLoader.m

```diff
@@ -0,0 +1,123 @@
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "PBFacebookBannerAdLoader.h"
+
+#import "PBConstants.h"
+
+struct FBAdSize {
+    CGSize size;
+};
+
+@implementation PBFacebookBannerAdLoader
+
+- (void)loadAd:(NSDictionary *)info {
+	NSString *bidPayload = (NSString *)info[@"adm"];
+	CGFloat height = [(NSString *)info[@"height"] floatValue];
+	CGSize adSize = CGSizeMake(-1, height);
+
+	// TODO nicole validate adSize against FBAdSize
+
+	// Load FBAdView using reflection so we can load the ad properly in the FBAudienceNetwork SDK
+	Class fbAdViewClass = NSClassFromString(@"FBAdView");
+	#pragma clang diagnostic push
+	#pragma clang diagnostic ignored "-Warc-performSelector-leaks"
+    if (fbAdViewClass != nil) {
+        SEL initMethodSel = NSSelectorFromString(@"initWithPlacementID:adSize:rootViewController:");
+        id fbAdViewObj = [fbAdViewClass alloc];
+        if ([fbAdViewObj respondsToSelector:initMethodSel]) {
+            NSMethodSignature *methSig = [fbAdViewObj methodSignatureForSelector:initMethodSel];
+            NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:methSig];
+
+            [invocation setSelector:initMethodSel];
+            [invocation setTarget:fbAdViewObj];
+
+            // Set arguments for init method
+            NSString *placementId = [self parsePlacementIdFromBidPayload:bidPayload];
+            struct FBAdSize fbAdSize;
+            fbAdSize.size = adSize;
+            UIViewController *vc = (UIViewController *)[NSObject new];
+
+            [invocation setArgument:&placementId atIndex:2];
+            [invocation setArgument:&fbAdSize atIndex:3];
+            [invocation setArgument:&vc atIndex:4];
+
+            // Invoke init method and use temp result to avoid crash later
+            [invocation invoke];
+            id __unsafe_unretained tempResultSet;
+            [invocation getReturnValue: &tempResultSet];
+            id result = tempResultSet;
+
+            // Set selector variables for other methods we need to call on FBAdView
+            SEL setDelegateSel = NSSelectorFromString(kFBSetDelegateSelName);
+            SEL loadAdSel = NSSelectorFromString(kFBLoadAdWithBidPayloadSelName);
+            SEL disableAutoRefreshSel = NSSelectorFromString(@"disableAutoRefresh");
+
+            if ([result respondsToSelector:setDelegateSel] &&
+                [result respondsToSelector:loadAdSel] &&
+                [result respondsToSelector:disableAutoRefreshSel]) {
+                    // Set up FBAdView and loadAdWithBidPayload
+                    [result performSelector:setDelegateSel withObject:self];
+                    [result performSelector:disableAutoRefreshSel];
+                    UIWindow *window = [[UIApplication sharedApplication] keyWindow];
+                    UIView *topView = window.rootViewController.view;
```

Comment:
Is window and topView still being used? should the result be added to the view hierarchy like Android?